r/R/dplyr-funcs-datetime.R

```diff
@@ -147,5 +147,14 @@ register_bindings_datetime <- function() {
   register_binding("pm", function(x) {
     !call_binding("am", x)
   })
-
+  register_binding("semester", function(x, with_year = FALSE) {
+    month <- call_binding("month", x)
+    semester <- call_binding("if_else", month <= 6, 1L, 2L)
+    if (with_year) {
+      year <- call_binding("year", x)
+      return(year + semester / 10)
```

Comment:
😱 This looks totally wrong, but it isn't. It's exactly what lubridate's semseter/quarter returns: a numeric where the decimal portion is the quarter or semester number (and not a decimal representation of such, or a string).

Reply:
Yes, initially I implemented it as a string, as at least that felt like adding a guardrail, but nope, it's numeric.